### PR TITLE
Feature/fga/better media handling

### PR DESCRIPTION
**Pull request by ganfra**
This PR doesn't handle yet media cancellation, as I couldn't make it work with coroutine at the moment (more investigation is needed).
So it's mostly about handling sound media (upload, rendering, simple viewer like file).

Closes https://github.com/vector-im/element-x-android/issues/857

### CODE REVIEW

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemAudioView.kt [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                color = ElementTheme.materialColors.secondary,
+                style = ElementTheme.typography.fontBodySmRegular,
+                maxLines = 1,
+                overflow = TextOverflow.Ellipsis,
+            )
+        }
+    }
+}
+
+@Preview
```
Comment (jmartinesp) [inline features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemAudioView.kt:92]: Nit: we could use `@DayNightPreviews` here and have a single Preview function.

File: tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.features.messages.impl.actionlist_null_DefaultGroup_SheetContentPreview-D-0_1_null_5,NEXUS_5,1.0,en].png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.features.messages.impl.actionlist_null_DefaultGroup_SheetContentPreview-D-0_1_null_5,NEXUS_5,1.0,en].png:1]: The attachment icon has vanished.

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/ProgressDialog.kt [hunk 2]
```diff
@@ -81,6 +95,8 @@ sealed interface ProgressDialogType {
 private fun ProgressDialogContent(
     modifier: Modifier = Modifier,
     text: String? = null,
+    isCancellable: Boolean = true,
```
Comment (bmarty) [inline libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/ProgressDialog.kt:98]: Maybe set to false by default like the public fun?

File: tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.libraries.designsystem.components_null_Dialogs_ProgressDialogPreview_0_null,NEXUS_5,1.0,en].png
Comment (jmartinesp) [inline tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.libraries.designsystem.components_null_Dialogs_ProgressDialogPreview_0_null,NEXUS_5,1.0,en].png:1]: The paddings here seem a bit excessive.

**Review by bmarty — CHANGES_REQUESTED** [review]
There is a regression in the screenshot test, else did you repro https://github.com/vector-im/element-x-android/issues/857 during your test?

File: libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/ProgressDialog.kt (hunk 2 above)
Comment (ganfra) [reply libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/components/ProgressDialog.kt:98]: yes good catch.

File: tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.libraries.designsystem.components_null_Dialogs_ProgressDialogPreview_0_null,NEXUS_5,1.0,en].png
Comment (ganfra) [reply tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.libraries.designsystem.components_null_Dialogs_ProgressDialogPreview_0_null,NEXUS_5,1.0,en].png:1]: If I'm not mistaken, it's the padding (24dp) used by default on M3 Dialog between content and buttons

Comment (bmarty) [reply tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.libraries.designsystem.components_null_Dialogs_ProgressDialogPreview_0_null,NEXUS_5,1.0,en].png:1]: The bottom padding looks excessive, yes, but it's similar on what we have elsewhere:

image

Comment (ganfra) [reply tests/uitests/src/test/snapshots/images/io.element.android.tests.uitests_ScreenshotTest_preview_tests[io.element.android.libraries.designsystem.components_null_Dialogs_ProgressDialogPreview_0_null,NEXUS_5,1.0,en].png:1]: Don't forget this is also temporary UI, so lets not worry about it for now.

**Review by bmarty — APPROVED** [review]
Thanks for the update.

**Pull request merged**